src/redux/books.js/booksSlice.js

```diff
@@ -0,0 +1,44 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  bookItem: [
+    {
+      id: 'item1',
+      title: 'Introduction to cybersecurity',
+      author: 'john Dow',
+      category: 'Fiction',
+    },
+    {
+      id: 'item2',
+      title: 'Secrete of success',
+      author: 'Napolioon Hill',
+      category: 'Nonfiction',
+    },
+    {
+      id: 'item3',
+      title: 'art of war',
+      author: 'muh Ndlovu',
+      category: 'Nonfiction',
+    },
+  ],
+};
+const booksSlice = createSlice({
+  name: 'book',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.bookItem.push(action.payload);
+    },
```

Comment:
- [OPTIONAL] Your work is great. I only provide this as an alternative and great way too.

```
export const booksSlice = createSlice({
  name: "books",
  initialState: {
    books: [],
  },
  reducers: {
    addBook: (state, action) => {
      state.books = [
        ...state.books,
        {
          id: Math.floor(Math.random() * 100),
          title: action.payload.title,
          author: action.payload.author,
        },
      ]
    }
```